client/verta/verta/client.py

```diff
@@ -1735,3 +1738,23 @@ def find_datasets(
             datasets = datasets.find(predicates)
 
         return datasets
+
+    def _create_workspace(self, org_id, workspace_name, resource_action_groups):
+        response_groups = self._conn.make_proto_request(
+            "GET", "/api/v1/uac-proxy/organization/{}/groups".format(org_id)
+        )
+        response_roles = self._conn.make_proto_request(
+            "GET", "/api/v2/uac-proxy/organization/{}/roles".format(org_id)
+        )
+        groups = self._conn.maybe_proto_response(response_groups, _Group.SearchGroups.Response).groups
+        all_users_group_id = next(iter(set(group.id for group in groups if group.name == "All Users")))
+        admins_group_id = next(iter(set(group.id for group in groups if group.name == "Admins")))
+        roles = self._conn.maybe_proto_response(response_roles, _Role.SearchRolesV2.Response).roles
+        super_user_role_id = next(iter(set(role.id for role in roles if role.name == "Super User")))
```

Comment:
There is a lot of work happening here, namely 4 or 5 different requests that all seem only tangentially related to the operation of creating the workspace itself.  Perhaps `Groups` should have its own class and set of helper functions for returning these various subsets of users.

Something along the lines of 
```
        groups: List[str] = Groups._find_proto(
            self.conn,
            workspace_name,
            org_id,
            )
        custom_role_id: int = Role._create(
            self._conn,
            workspace_name + "role", 
            org_id, 
            resource_action_groups
            ).id
        return Workspace._create(
            self._conn,
            workspace_name,
            org_id,
            [
                _Workspace.Permission(
                    group_id = groups.admins,
                    role_id = groups.supers,
                    ),
                _Workspace.Permission(
                    group_id = groups.all,
                    role_id = custom_role_id,
                    )
            ]
        )
```
Just a thought, as this client file is already a bit unwieldy

Reply:
As an alternative, just as a bare minimum to pull this logic out of `client.py`, I think we could have a

```python
class Organization:  # has its own _conn and id
    ...

    def create_role(name, resource_actions):
        return Role._create(...)

    def get_groups():
        ...
        return {group.name: group for group in groups}

    def get_roles():
        ...
        return {role.name: role for role in roles}
```

That way we do not have to keep passing around `self._conn` and `org_id`, too.

Reply:
added Organization

Reply:
Can we consider just expanding the existing `verta.tracking._organization.Organization()` class instead of creating a secondary `v2` class for Organization?  It would be better to have a single class for organization to avoid confusion

Reply:
existing Organization in main is used in tests. I think old organization in this PR is not used anymore except 3 CollaboratorType usages which can be easily removed. So I think existing Organization is a candidate for removal and not sure that we need to combine it with the new organization. Also we haven't been adding any backward compatibility for PythonClient that is why I'm not sure that existing Organization needed. OrganizationV2 can be renamed to Organization. I just renamed it to V2 to be able to create a separate change: https://github.com/VertaAI/modeldb/pull/3497

Reply:
That's a fair question @ewagner-verta. _Technically_ they could be combined. But _pragmatically_, I think what we want is to remove the original `_organization` `Organization` (it was never meant for public use anyway) and only use the new `_uac` `OrganizationV2`.

But like @evgeniilezhnin mentioned, we might still need the old one for compatibility with some existing tests. As soon as we determine it's not needed, `_organization` `Organization` should be removed.